### PR TITLE
feat: minting function

### PR DESCRIPTION
**Pull request by nekaartajaya**

### CODE REVIEW

File: .env-template [hunk 1]
```diff
@@ -3,4 +3,5 @@ NEXT_PUBLIC_API_URL=
 NEXT_PUBLIC_WEB3_PROVIDER_URL=
 NEXT_PUBLIC_WEB3_PROVIDER_KEY=
 NEXT_PUBLIC_LBSF_CONTRACT_ADDRESS=
-NEXT_PUBLIC_USDC_CONTRACT_ADDRESS=
\ No newline at end of file
+NEXT_PUBLIC_USDC_CONTRACT_ADDRESS=
+NEXT_PUBLIC_CHAIN_ID=
```
Comment (irmannmal) [inline .env-template:7]: why we need this?

File: src/helpers/metamask-interact.ts [hunk 2]
```diff
@@ -51,15 +51,17 @@ export const getNFTImage = async () => {
 
     const uri = await contract.methods.uri(activeStage).call();
 
-    const resp = await fetch(uri);
+    const resp = await fetch(
+      uri.replace('ipfs://', 'https://gateway.pinata.cloud/ipfs/'),
```
Comment (irmannmal) [inline src/helpers/metamask-interact.ts:55]: please move this to helper function, to avoid duplication code

**Review by irmannmal — CHANGES_REQUESTED** [review]

File: .env-template (hunk 1 above)
Comment (nekaartajaya) [reply .env-template:7]: next_public_chain_id is for checking the right network for minting

File: src/helpers/metamask-interact.ts (hunk 2 above)
Comment (nekaartajaya) [reply src/helpers/metamask-interact.ts]: done

**Comment by irmannmal** [comment]
i'll change CI/CD pipeline for new env in the next PR @nekaartajaya

**Review by irmannmal — APPROVED** [review]

**Pull request merged**